File: frontend/packages/lib-shared/api/modules/adDownstreamMedia.ts
import type { CordysAxios } from '@lib/shared/api/http/Axios';
import { AdDownstreamMediaUrl, AdDownstreamMediaPageUrl } from '@lib/shared/api/requrls/adDownstreamMedia';
import type { CommonList } from '@lib/shared/models/common';
import type { AdDownstreamMediaAccount } from '@lib/shared/models/advertising';

export interface AdDownstreamMediaItem {
    id: string;
    name?: string;
    mediaType?: string;
    mediaTypeLabel?: string;
    channel?: string;
    rateCard?: string;
    discountPolicy?: string;
    contactPerson?: string;
    contactPhone?: string;
    cooperationStatus?: number;
    status?: number;
    statusLabel?: string;
    businessEntityId?: string;
    businessEntityName?: string;
    createTime?: number;
}

export interface AdDownstreamMediaDetail extends AdDownstreamMediaItem {
    accountList?: AdDownstreamMediaAccount[];
}

export interface AdDownstreamMediaSaveParams {
    id?: string;
    name?: string;
    mediaType?: string;
    channel?: string;
    rateCard?: string;
    discountPolicy?: string;
    contactPerson?: string;
    contactPhone?: string;
    cooperationStatus?: number;
    status?: number;
    businessEntityId?: string;
    accounts?: AdDownstreamMediaAccountSaveItem[];
}

export interface AdDownstreamMediaAccountSaveItem {
    id?: string;
    payeeName?: string;
    bankName?: string;
    bankAccount?: string;
    disabled?: number;
}

type AdDownstreamMediaPageParams = Record<string, any>;
type AdDownstreamMediaPageResult = CommonList<AdDownstreamMediaItem>;

export default function useAdDownstreamMediaApi(CDR: CordysAxios) {
    function createAdDownstreamMedia(data: AdDownstreamMediaSaveParams) {
        return CDR.post<any>({ url: AdDownstreamMediaUrl, data });
    }
    function updateAdDownstreamMedia(data: AdDownstreamMediaSaveParams) {
        return CDR.put<any>({ url: AdDownstreamMediaUrl, data });
    }
    function getAdDownstreamMediaDetail(id: string) {
        return CDR.get<AdDownstreamMediaDetail>({ url: `${AdDownstreamMediaUrl}/${id}` });
    }
    function getAdDownstreamMediaAccounts(id: string) {
        return CDR.get<AdDownstreamMediaAccount[]>({ url: `${AdDownstreamMediaUrl}/${id}/accounts` });
    }
    function getAdDownstreamMediaPage(data: AdDownstreamMediaPageParams) {
        return CDR.post<AdDownstreamMediaPageResult>({ url: AdDownstreamMediaPageUrl, data }, { ignoreCancelToken: true });
    }
    function deleteAdDownstreamMedia(id: string) {
        return CDR.delete({ url: `${AdDownstreamMediaUrl}/${id}` });
    }
    return { createAdDownstreamMedia, updateAdDownstreamMedia, getAdDownstreamMediaDetail, getAdDownstreamMediaAccounts, getAdDownstreamMediaPage, deleteAdDownstreamMedia };
}

File: backend/crm/src/main/java/cn/cordys/crm/ad/downstreammedia/domain/AdDownstreamMedia.java
package cn.cordys.crm.ad.downstreammedia.domain;

import cn.cordys.common.domain.BaseModel;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Table(name = "ad_downstream_media")
public class AdDownstreamMedia extends BaseModel {

    @Schema(description = "名称")
    private String name;

    @Schema(description = "类型")
    private String mediaType;

    @Schema(description = "覆盖渠道")
    private String channel;

    @Schema(description = "刊例价")
    private String rateCard;

    @Schema(description = "折扣政策")
    private String discountPolicy;

    @Schema(description = "联系人")
    private String contactPerson;

    @Schema(description = "电话")
    private String contactPhone;

    @Schema(description = "合作状态:10正常/20停用")
    private Integer cooperationStatus;

    @Schema(description = "状态:10正常/20停用")
    private Integer status;

    @Schema(description = "业务主体")
    private String businessEntityId;

    @Schema(description = "组织(租户)id")
    private String organizationId;

    @Schema(description = "是否删除:0-否/1-是")
    private Integer deleted = 0;
}
